### PR TITLE
update readme docs

### PR DESCRIPTION
**Pull request by csantanapr**
updates for main README about native dev
updates for vagrant README more clear instructions
updates for macos README to fix remove of pipe and bash

### CODE REVIEW

File: tools/vagrant/README.md [hunk 1]
```diff
@@ -106,37 +110,36 @@ cd ${OPENWHISK_HOME}
 ```
 
 ### Safe Re-deploy (after VM restart)
-Use ansible to deploy from inside the VM, this is done automatically once at VM creation.
-You can deploy OpenWhisk and keep the last state of the data store.
+You can re-deploy OpenWhisk and keep the last state of the data store.
 This will keep the transient data store including user actions and sytem catalog, even if using CouchDB container
-You can do this after restarting the Vagrant VM using `vagrant reload` or after starting the VM with `vagrant up` from halt or suspended state
+If you restart the VM, OpenWhisk containers will be running, but in a bad state.
+The following commands are helpful after restarting the VM using `vagrant reload` or after starting the VM with `vagrant up` from halt or suspended state
```
Comment (rabbah) [inline tools/vagrant/README.md]: halt -> halted?

File: tools/vagrant/README.md [hunk 2]
```diff
@@ -79,6 +79,8 @@ wsk -i action invoke /whisk.system/samples/echo -p message hello --blocking --re
 configure the CLI with new values for __apihost__, __namespace__, and __auth__ key.
  
 ### Use the wsk CLI inside the VM
+Inside the VM the CLI command `wsk` points to `$HOME/bin/wsk` this is a wrapper script that uses `$OPENWHISK_HOME/bin/go-cli/linux/amd64/wsk` and the flag `-i` to be able to connect insecurly.
```
Comment (rabbah) [inline tools/vagrant/README.md]: For your convenience, a `wsk` wrapper is provided inside the VM which delegates CLI commands to ... and adds the `-i` parameter that is required for insecure access to the local OpenWhisk deployment.

**Comment by rabbah** [comment]
If possible, can we address #1245 as well in this pr? Or do you prefer a different PR?

**Comment by csantanapr** [comment]
>If possible, can we address #1245 as well in this pr? Or do you prefer a different PR?

I will address with this one, but I have to admin I think something changed in github handling links in markdown this used to work.

**Comment by csantanapr** [comment]
@rabbah please review 2nd commit https://github.com/openwhisk/openwhisk/pull/1240/commits/87f9102cb9bf235c4b8f75be0573a96dba8567a8

There is no need to do changes for #1245 it looks like it self fixed b github backend

**Comment by rabbah** [comment]
Edited lightly and merged as https://github.com/openwhisk/openwhisk/commit/c76cfc8eda99638966dca5284f36bf392c9cf757.

**Pull request closed without merging**